### PR TITLE
Add synced_collections

### PR DESCRIPTION
**Pull request by vyasr**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by bdice** [comment]
I am willing to be listed as a maintainer for this package. Thanks @vyasr!

File: recipes/synced_collections/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  license_family: BSD
+  license_file: LICENSE.txt
+  summary: Interact with persistent key-value stores using Pythonic abstractions.
+  description: |
+    The signac framework helps users manage and scale file-based workflows,
+    facilitating data reuse, sharing, and reproducibility. It provides a simple
+    and robust data model to create a well-defined indexable storage layout for
+    data and metadata. This makes it easier to operate on large data spaces,
+    streamlines post-processing and analysis and makes data collectively
+    accessible.
```
Comment (bdice) [inline recipes/synced_collections/meta.yaml:46]: Can we rewrite this? Proposing this:

> The synced_collections package provides Pythonic abstractions over various underlying data stores, presenting APIs that behave like standard built-in Python collections like dicts.

**Comment by vyasr** [comment]
@conda-forge-admin please lint

File: recipes/synced_collections/meta.yaml (hunk 1 above)
Comment (vyasr) [reply recipes/synced_collections/meta.yaml:46]: Yes I'm fine with rewriting. Happy to use something like this.

**Comment by b-butler** [comment]
I am willing to be a maintainer as well.

File: recipes/synced_collections/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    of data stores, including MongoDB databases and Zarr groups.
+  doc_url: https://docs.signac.io
+  dev_url: https://github.com/glotzerlab/synced_collections
+
+extra:
+  recipe-maintainers:
+    - b-butler
+    - bdice
+    - csadorf
+    - mikemhenry
```
Comment (vyasr) [inline recipes/synced_collections/meta.yaml:53]:
```suggestion
```

**Comment by vyasr** [comment]
@conda-forge/help-python, ready for review!

**Comment by mikemhenry** [comment]
I am willing to be a maintainer as well.

**Review by trungleduc — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 🙏

**Pull request merged**